src/dune/foreign_sources.ml

```diff
@@ -159,19 +150,36 @@ let make (d : _ Dir_with_dune.t) ~(sources : Foreign.Sources.Unresolved.t) =
         (snd (List.hd exes.names), m))
   in
   let () =
-    let rev_map =
-      List.concat_map libs ~f:(fun (_, c_sources) ->
-          String.Map.values c_sources
-          |> List.map ~f:(fun (loc, source) ->
-                 (Foreign.Source.path source, loc)))
-      |> Path.Build.Map.of_list
+    let objects =
+      List.concat
+        [ List.map libs ~f:snd
+        ; List.map foreign_libs ~f:(fun (_, (_, sources)) -> sources)
+        ; List.map exes ~f:snd
+        ]
+      |> List.concat_map ~f:(fun sources ->
+             String.Map.values sources
+             |> List.map ~f:(fun (loc, source) ->
+                    let object_path =
+                      Foreign.Source.path source |> Path.Build.split_extension
+                      |> fst |> Path.build |> Path.drop_optional_build_context
+                      |> Path.to_string
+                    in
```

Comment:
I changed this because `Path.to_string` produces root-relative paths, which seems worse than dune-file-relative, and becomes simpler.